packages/components/src/text-control/types.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import type { HTMLInputTypeAttribute } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { BaseControlProps } from '../base-control/types';
+import type { WordPressComponentProps } from '../ui/context';
+
+export type TextControlProps = WordPressComponentProps<
+	Pick<
+		BaseControlProps,
+		'className' | 'hideLabelFromVision' | 'help' | 'label'
+	> & {
+		/**
+		 * A function that receives the value of the input.
+		 */
+		onChange: ( value: string ) => void;
+		/**
+		 * The current value of the input.
+		 */
+		value: string | number;
```

Comment:
If I were to write this component from scratch, I would type `value` as a `string`.

Since this component is _not_ experimental (and since the purpose of these TypeScript migrations is to introduce the least amount of changes possible), we may want to keep this type as it currently is, but consider updating it in a follow-up (see also this recent related [conversation](https://github.com/WordPress/gutenberg/pull/40535#issuecomment-1112049865))

Reply:
I agree it should be `string` in this case. The old jsdocs said `string` but the README file says `string | number`.

Reply:
Let's keep it as `string|number` for this PR. It would be great if we could then open a follow-up PR to change it to be only `string` — we could then assess in that PR if effectively it's a change that can be applied without introducing any breaking change.

Would you be up for that?